main.tf

```diff
@@ -8,11 +8,11 @@ module "ssh-key" {
 }
 
 resource "azurerm_kubernetes_cluster" "main" {
-  name                    = "${var.prefix}-aks"
+  name                    = var.cluster_name == "" ? "${var.prefix}-aks" : var.cluster_name
   kubernetes_version      = var.kubernetes_version
   location                = data.azurerm_resource_group.main.location
   resource_group_name     = data.azurerm_resource_group.main.name
-  dns_prefix              = var.prefix
+  dns_prefix              = var.cluster_name == "" ? var.prefix : var.cluster_name
```

Comment:
Here could we keep using `var.prefix` as `dns_prefix`? The user may not want `dns_prefix` the same as the `cluster_name`?

Reply:
yes, it makes sense. I am going to edit this.

Reply:
`dns_prefix              = var.prefix ? var.prefix : var.cluster_name`